kong/api/routes/ca_certificates.lua

```diff
@@ -0,0 +1,56 @@
+local ngx = ngx
+local kong = kong
+local ipairs = ipairs
+local null = ngx.null
+local string_format = string.format
+
+
+local function each_service(entity)
+  local options = {
+    workspace = null,
+  }
+
+  local iter = entity:each(1000, options)
+  local function iterator()
+    local element, err = iter()
+    if err then return nil, err end
+    if element == nil then return end
+    if element.ca_certificates and #element.ca_certificates > 0 then return element, nil end
+    return iterator()
+  end
+
+  return iterator
+end
+
+
+local function verify_references(id)
+  -- services
+  for service, err in each_service(kong.db.services) do
+    if err then
+      ngx.log(ngx.ERR, "could not load services: ", err)
```

Comment:
localize `ngx.log` `ngx.ERR`

Reply:
Why should rarely called functions be localized?  There is no benefit, but inventing extra names makes the code harder to review and refactor.

Reply:
use `kong.log.err(...` is better?